data_steward/cdr_cleaner/manual_cleaning_rules/dc732_suppress.py

```diff
@@ -0,0 +1,445 @@
+    Note: Rows are appended if tables exist
+    
+    :param tables_to_suppress_df: dataframe with dataset_id, table_name, ext_table_name, columns
+    :param suppress_rows_table: identifies the table where suppress rows are loaded
+    :return: list of query jobs
+    """
+    tables_group_df = tables_to_suppress_df.groupby(by=['table_name', 'ext_table_name'])
+    query_jobs = []
+    for (table_name, ext_table_name), table_info in tables_group_df:
+        dataset_ids = table_info.dataset_id.unique()
+        if table_name != 'observation':
+            backup_rows_dest_table = f'{PROJECT_ID}.{SANDBOX_DATASET_ID}.{ISSUE_PREFIX}{table_name}'
+            job_config = bigquery.QueryJobConfig(destination=backup_rows_dest_table)
+            job_config.write_disposition = 'WRITE_APPEND'
+            backup_table_rows_query = BACKUP_TABLE_ROWS_QUERY.render(dataset_ids=dataset_ids, table=table_name, suppress_rows_table=suppress_rows_table)
+            backup_table_query_job = CLIENT.query(backup_table_rows_query, job_config=job_config)
+            query_jobs.append(backup_table_query_job)
+            
+            if ext_table_name:
+                backup_ext_dest_table = f'{PROJECT_ID}.{SANDBOX_DATASET_ID}.{ISSUE_PREFIX}{ext_table_name}'
+                job_config = bigquery.QueryJobConfig(destination=backup_ext_dest_table)
+                job_config.write_disposition = 'WRITE_APPEND'
+                backup_ext_table_rows_query = BACKUP_EXT_TABLE_ROWS_QUERY.render(dataset_ids=dataset_ids, table=table_name, suppress_rows_table=suppress_rows_table)
+                backup_ext_table_query_job = CLIENT.query(backup_ext_table_rows_query, job_config=job_config)
+                query_jobs.append(backup_ext_table_query_job)
+    return query_jobs
+
+
+# -
+
+# # Execute Deletion
+# Use a meta-query to generate the minimal set of DELETE statements needed to delete rows in domain and \_ext tables.
+
+# +
+GET_DELETE_QUERIES = JINJA_ENV.from_string("""
+WITH table_ids AS
+(SELECT DISTINCT dataset_id, table 
+FROM `{{suppress_rows_table}}`)
+SELECT CONCAT('DELETE FROM `', 
+ dataset_id, '.', table, '` t', 
+ ' WHERE EXISTS', 
+ '(SELECT 1 FROM {{suppress_rows_table}} WHERE dataset_id = "', dataset_id, '" AND table = "', table, '" AND t.', table, '_id = row_id)')
+AS query_string
+FROM table_ids
+""")
+
+def get_delete_queries(suppress_rows_table):
+    """
+    Get list of DELETE statements needed to delete rows in domain and _ext tables
+    
+    :param suppress_rows_table: identifies the table where suppress rows are loaded
+    """
+    get_delete_query = GET_DELETE_QUERIES.render(suppress_rows_table=suppress_rows_table)
+    delete_query_job = CLIENT.query(get_delete_query)
+    delete_query_results = list(delete_query_job.result())
+    all_delete_queries = []
+    for delete_query_result in delete_query_results:
+        delete_query = delete_query_result['query_string']
+        # HACK to get associated _ext DELETE statement
```

Comment:
nit:  Is this really a hack?  Maybe just have a nice comment describing what you are doing.